code/game/objects/items/cigs_lighters.dm

```diff
@@ -194,12 +197,41 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	icon_state = icon_off
 	inhand_icon_state = inhand_icon_off
 
+	// "It is called a cigarette"
+	AddComponent(/datum/component/edible,\
+		initial_reagents = list_reagents,\
+		food_flags = null,\
+		foodtypes = JUNKFOOD,\
+		volume = 50,\
+		eat_time = 0 SECONDS,\
+		tastes = list("A never before experienced flavour.", "Finally sitting down after standing your entire life"),\
+		eatverbs = list("Chomps"),\
+		bite_consumption = 50,\
+		junkiness = 0,\
+		reagent_purity = null,\
+		on_consume = CALLBACK(src, PROC_REF(on_consume)),\
+	)
+
 /obj/item/cigarette/Destroy()
 	STOP_PROCESSING(SSobj, src)
 	QDEL_NULL(mob_smoke)
 	QDEL_NULL(cig_smoke)
 	return ..()
 
+/obj/item/cigarette/proc/on_consume(mob/living/eater, mob/living/feeder)
+	if(isnull(eater.client))
+		return
+	var/ckey = eater.client.ckey
+	// We must have more!
+	GLOB.cigarette_eaters[ckey]++
+	if(GLOB.cigarette_eaters[ckey] >= 500)
+		eater.client.give_award(/datum/award/achievement/misc/cigarettes)
+
+/obj/item/cigarette/examine(mob/user)
+	. = ..()
+	if(!is_species(user, /datum/species/human))
```

Comment:
Why?

Reply:
Simply because there's no helper for it (Also I don't think a human would wish to taste it)

Reply:
Add another check for the TRAIT_NOHUNGER trait. That's all.

Reply:
On it sir